editor.planx.uk/src/@planx/components/FindProperty/Public/index.tsx

```diff
@@ -124,11 +110,62 @@ function Component(props: Props) {
     }
   }, [data, address]);
 
-  useEffect(() => {
-    if (geojson) setBoundary(geojson);
-  }, [geojson]);
+  const validateAndSubmit = () => {
+    // TODO `if (isValidating)` on either page, wrap Continue button in error mesage?
+
+    if (page === "new-address") {
+      if (address?.x === undefined && address?.y === undefined)
+        setMapValidationError("Click or tap to place a point on the map");
+
+      if (address?.title === undefined) setShowSiteDescriptionError(true);
+    }
+
+    if (address) {
+      const newPassportData: Store.userData["data"] = {};
+      newPassportData["_address"] = address;
+      if (address?.planx_value) {
+        newPassportData["property.type"] = [address.planx_value];
+      }
+
+      if (localAuthorityDistricts) {
+        newPassportData["property.localAuthorityDistrict"] =
+          localAuthorityDistricts;
+      }
+      if (regions) {
+        newPassportData["property.region"] = regions;
+      }
+      if (titleBoundary) {
+        const areaSquareMetres =
+          Math.round(area(titleBoundary as Feature) * 100) / 100;
+        newPassportData["property.boundary.title"] = titleBoundary;
+        newPassportData["property.boundary.title.area"] = areaSquareMetres;
+        newPassportData["property.boundary.title.area.hectares"] =
+          squareMetresToHectares(areaSquareMetres);
+      }
+
+      newPassportData[PASSPORT_COMPONENT_ACTION_KEY] =
+        address?.source === "os"
+          ? FindPropertyUserAction.Existing
+          : FindPropertyUserAction.New;
+
+      props.handleSubmit?.({ data: { ...newPassportData } });
+    }
+  };
+
+  return (
+    <Card
+      handleSubmit={validateAndSubmit}
+      isValid={
+        page === "new-address" && !isValidating
+          ? true
+          : Boolean(address) && !isValidating
+      }
+    >
+      {getBody()}
```

Comment:
nit: This was a bit unexpected - it should really be a `React.FC`, e.g. `<Body/>` (or some other name).

I know this isn't introduced here so it can be ignored or fixed forward as you feel best - same applies to DrawBoundary

Reply:
Thanks for noting this - will fix forward in this case to not over-complicate dependent PR right now !